### DISCUSSION TITLE
Can duckdb read/query it's own persisted file from s3?

### DISCUSSION THREAD

**Discussion by pmg1991** [duckdb/duckdb#8683] [General]
Let's say there is one file data.db created locally but uploaded to s3 bucket. Can httpfs plugin read this file? Will data it be read as per query or whole file will be downloaded. Please provide a snippet that would be helpful.

**Comment by Mause** [comment]
You can use `ATTACH` to read a remote database yes, but only in read only mode.

Examples are here: https://duckdb.org/docs/archive/0.8.1/sql/statements/attach#examples
